### DISCUSSION TITLE
访问加密文件夹 500 错误

### DISCUSSION THREAD

**Discussion by trymesoft** [spencerwooo/onedrive-vercel-index#570] [Q&A] [answered]
## 操作流程

1. 按照文档，修改 config/site.config.js 中的 protectedRoutes 字段，添加需要加密的文件夹绝对路径
   ```
   protectedRoutes: ['/🔐 Private Folder/📦 Sensitive Things', '/🔐 Test Files']
   ```
   在相应文件夹中已建立 `.password` 文件。
2. 访问加密文件夹报错，控制台报错 `500`，对应 API 返回值为：

   `{"error":"Internal server error."}`

3. 对应截图：
![image](https://user-images.githubusercontent.com/24319553/160100105-d66a2506-f0cc-4b93-845d-feea80353447.png)

## 尝试操作

1. 清空 UpStash Redis 缓存的 access_token 及 refresh_token，重新授权，错误同上；
2. 尝试修改路径名称，错误同上；
3. 本地自编译，启动后错误同上；
4. 由于文档说明需要用绝对路径，但未指明是否是加上 `baseDirectory` 配置的绝对路径，两种方式均尝试过，错误同上。

**Comment by spencerwooo** [comment]
刚看了下你的仓库我觉得应该没什么问题？加密文件夹的路径是相对你的 `/Public` 路径的，你设置的应该没什么问题。本地编译后服务端的报错信息有看到吗？

**Reply by onmoso** [reply]
测试发现，编码的问题可能性比较大。打开文件另存为Unicode编码可行，其他编码未测试。

**Reply by trymesoft** [reply]
> 测试发现，编码的问题可能性比较大。打开文件另存为Unicode编码可行，其他编码未测试。

编码为 UTF-8

**Reply by code004Accepted** [reply]
这边遇到了同样的问题[#605](https://github.com/spencerwooo/onedrive-vercel-index/discussions/605)，站点中所有`.password`文件都是UTF-8，换行符是CRLF，但其他的都可以正常工作，唯独这个不行。
唯一的区别是其他密码是包含非数字字符的字符串，而这个密码只包含数字。或许是这个问题？

**Comment by pseudoxx** [comment]
遇到了同样的问题 泪目

**Reply by pseudoxx** [reply]
> 遇到了同样的问题 泪目

是世纪互联企业版

**Comment by Gitnapp** [comment]
遇到了同样的问题 泪目 +1

**Comment by code004Accepted** [comment, marked as answer]
See #606
